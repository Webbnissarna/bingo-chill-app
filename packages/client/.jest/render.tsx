import React from "react";
import { render as rtlRender } from "@testing-library/react";

export function render(ui: React.ReactElement) {
  return rtlRender(<>{ui}</>);
}
