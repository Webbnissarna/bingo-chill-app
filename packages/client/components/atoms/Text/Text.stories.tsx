import type { Story, StoryDefault } from "@ladle/react";
import Text from ".";

interface StoryProps {
  text: string;
}

function StoryComponent({ text }: StoryProps) {
  return <Text>{text}</Text>;
}

export const Default: Story<StoryProps> = StoryComponent;

Default.args = {
  text:
    "Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis repellat id" +
    "sed esse, numquam itaque ex aperiam provident quaerat deserunt sunt, harum" +
    "temporibus non. Quae ipsum quas nobis illum rerum.",
};

Default.argTypes = {
  text: {
    control: { type: "text" },
  },
};

export default {
  title: "Atoms / Text",
} satisfies StoryDefault;
