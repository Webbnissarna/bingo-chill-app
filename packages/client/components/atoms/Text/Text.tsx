interface TextProps {
  children: string | number;
}

export default function Text({ children }: TextProps): JSX.Element {
  return (
    <span className="font-text text-snowStorm-2 text-base">{children}</span>
  );
}
