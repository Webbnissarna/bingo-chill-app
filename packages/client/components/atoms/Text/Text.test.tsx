import { render } from "../../../.jest/render";
import Text from ".";

describe("Text", () => {
  it("renders", () => {
    const { asFragment } = render(
      <Text>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Magnam
        voluptatibus officia voluptate dolores esse maiores omnis veniam harum
        pariatur animi vitae, numquam, nesciunt, velit sed blanditiis
        cupiditate? Cum, quaerat commodi.
      </Text>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
