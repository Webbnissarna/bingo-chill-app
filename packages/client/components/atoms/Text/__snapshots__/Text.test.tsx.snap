// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders 1`] = `
<DocumentFragment>
  <span
    class="font-text text-snowStorm-2 text-base"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Magnam voluptatibus officia voluptate dolores esse maiores omnis veniam harum pariatur animi vitae, numquam, nesciunt, velit sed blanditiis cupiditate? Cum, quaerat commodi.
  </span>
</DocumentFragment>
`;
